src/components/DotIndicatorMessage.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import _ from 'underscore';
+import PropTypes from 'prop-types';
+import {View} from 'react-native';
+import styles from '../styles/styles';
+import Icon from './Icon';
+import * as Expensicons from './Icon/Expensicons';
+import colors from '../styles/colors';
+import variables from '../styles/variables';
+import Text from './Text';
+
+const propTypes = {
+    /** The messages to display  */
+    messages: PropTypes.oneOfType([
+        PropTypes.arrayOf(PropTypes.string),
+        PropTypes.objectOf(PropTypes.string),
+    ]),
+
+    /** The type of message, 'error' shows a red dot, 'success' shows a green dot */
+    type: PropTypes.oneOf(['error', 'success']).isRequired,
+
+    /** Additional styles to apply to the container */
+    // eslint-disable-next-line react/forbid-prop-types
+    style: PropTypes.arrayOf(PropTypes.object),
+};
+
+const defaultProps = {
+    messages: [],
+    style: [],
+
+};
+
+const DotIndicatorMessage = (props) => {
+    if (_.isEmpty(props.messages)) {
+        return null;
+    }
+
+    // If the messages are passed as an object (such as our Onyx errors), then grab the sorted values
+    let sortedMessages = props.messages;
+    if (!_.isArray(props.messages)) {
```

Comment:
this feels like "data massaging" which we don't want but I'm not super familiar with how this is all being used. We want data in onyx to be used as-is right so if there is something that isn't passing it's data to this component properly then let's fix that - I'm fine not blocking this PR on fixing that so feel free to also open a new GH if that's the case.

Reply:
hmm, I don't think this qualifies as that kind of "data massaging" we want to avoid:
- This component doesn't know that these `messages` come from onyx
- In onyx we store errors in an object many times keyed by time.. I'm not sure if it is avoidable to do this kind of sorting to be sure we display them in the intended order.

Reply:
yeah I agree with most of what you're saying. 

But what I'm getting at is instead of converting this to an array then we should just build it to assume it's coming from onyx and there for is an object. 

Anywhere this component is being used that isn't passing an object/data directly from onyx then should be updated so that it passes an object like `{microtime: 'error_mesage'}` instead of `['error_message']`.

Reply:
When will messages be an array? Isn't it always object? Also, kind of odd to check for !array instead of checking if it is an object, no?

Reply:
I see your point, It makes sense to me to require the shape to be `{microtime: 'error_mesage'}` to avoid having multiple possible ways of passing errors.

Reply:
okay i chatted with Yuwen and Marc as well and confirmed that data from onyx should just be an object of strings and update this. 

tested and confirmed both success and error cases work

<img width="1208" alt="Screen Shot 2022-08-31 at 3 08 31 PM" src="https://user-images.githubusercontent.com/4073354/187686483-0cdc935d-dc61-4150-a6a8-afc96e0e7cac.png">
<img width="1202" alt="Screen Shot 2022-08-31 at 3 07 53 PM" src="https://user-images.githubusercontent.com/4073354/187686496-7f8f38e7-9dec-4a68-9f61-0d875a9207dc.png">

confirmed no errors in my js console.

Reply:
Ohhh interesting, the reason I let it accept both objects and arrays in the propTypes was because I actually pass the messages as an array [here](https://github.com/Expensify/App/pull/10556/files#diff-c075eec38f8adc854c238b1d4cfeba1ed3e5de0aa740aa78a658ffe3a53814f9R75-R78) since I don't know what I'd key them with if I passed them as an object. That's a use-case where we don't pass them objects directly from Onyx like we would in the usage in `OfflineWithFeedback`.

Surprised it doesn't give you any propTypes errors, but hey if it works it works 🤷

Reply:
hmm yeah nope I was wrong I was getting an error that an array was passed when we expected object (I thought that was an old message when I first saw it).

need to adjust this slightly then as well. another push coming.